temporaltest/server_test.go

```diff
@@ -264,38 +281,74 @@ func TestClientWithCustomInterceptor(t *testing.T) {
 	}
 }
 
-func TestSearchAttributeCacheDisabled(t *testing.T) {
-	// TODO(jlegrone) re-enable this test when advanced visibility is enabled in temporalite.
-	t.Skip("This test case does not currently pass as of the 1.20 release")
-
+func TestSearchAttributeRegistration(t *testing.T) {
 	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
 	defer cancel()
 	ts := temporaltest.NewServer(temporaltest.WithT(t))
+	c := ts.GetDefaultClient()
 
-	testSearchAttr := "my-search-attr"
+	testSearchAttr := "MySearchAttr"
 
 	// Create a search attribute
-	_, err := ts.GetDefaultClient().OperatorService().AddSearchAttributes(ctx, &operatorservice.AddSearchAttributesRequest{
+	if _, err := ts.GetDefaultClient().OperatorService().AddSearchAttributes(ctx, &operatorservice.AddSearchAttributesRequest{
 		SearchAttributes: map[string]enums.IndexedValueType{
 			testSearchAttr: enums.INDEXED_VALUE_TYPE_KEYWORD,
 		},
 		Namespace: ts.GetDefaultNamespace(),
+	}); err != nil {
+		t.Fatal(err)
+	}
+	// Confirm search attribute is registered immediately
+	// TODO(jlegrone): investigate why custom search attribute missing here while setting it from workflow succeeds.
```

Comment:
I think there's a cache that you need to wait for it to refresh. If this works reliably as-is, I think it's fine and we can improve it later.

Reply:
This is with `system.forceSearchAttributesCacheRefreshOnRead` enabled by default in `TestServer`. Are there any other options you're aware of which we could try?

This test case has existed in the Temporalite project for a while: https://github.com/temporalio/temporalite/blob/2df426ad3426fb6ecc91d3b2e03274018bf14721/temporaltest/server_test.go#L200-L223

I suspect there might have been a regression in one of the recent releases, but haven't had a chance to dig into it further yet.

Reply:
Hmm, I'm not sure. @rodrigozhou do you know if this is using the same cache or why this might fail?

Reply:
There was a regression at #4017 was recently resolved (we held off updating Python and .NET ephemeral servers due to this regression).